dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.42/V2_42_25__Migrate_event_change_logs.sql

```diff
@@ -0,0 +1,60 @@
+-- DHIS2-18117: migrate tracker event change logs from trackedentitydatavalueaudit to eventchangelogs
+
+-- Create new table and add constraints
+create sequence if not exists eventchangelog_sequence;
+
+create table if not exists eventchangelog (
+    eventchangelogid int8 not null default nextval('eventchangelog_sequence'),
+    eventid int8 not null,
+    dataelementid int8 null,
+    eventproperty varchar(100) null,
+    currentvalue varchar(50000) null,
+    previousvalue varchar(50000) null,
+    changelogtype varchar(100) not null,
+    created timestamp not null,
+    createdby varchar(255) not null,
+    constraint eventchangelog_pkey primary key (eventchangelogid),
+    constraint fk_eventchangelog_dataelementid foreign key (dataelementid) references dataelement(dataelementid),
+    constraint fk_eventchangelog_eventid foreign key (eventid) references event(eventid)
+);
+
+create index if not exists index_eventchangelog_eventid on eventchangelog using btree (eventid);
```

Comment:
Can we also add the necessary indexes based on our query usage? I remember we are going to sort by created? So an index (or multiple index) to support combinations of filtering and sorting?

Reply:
We will support sorting by date, user, and data element/property, and will also allow filtering by these same attributes. I think it makes sense to refactor the existing index to a composite on `(eventid, created DESC)` since this is the default query. Even when sorting or filtering by other attributes this index will still be used because we will always join the query by `eventid`.

I'm not sure about adding more indexes, as this depends entirely on user parameters. Covering other possible combinations would mean indexing everything except the values and `changeLogType`, which could increase overhead and impact performance. 

Do you agree?